cc/private/compile/cc_compilation_helper.bzl

```diff
@@ -261,6 +262,12 @@ def _module_map_struct_to_module_map_content(parameters, tree_expander):
         add_header(path = header.path, visibility = "", can_compile = False)
         added_paths.add(header.path)
 
+    for header in parameters.public_textual_headers:
+        if header.path in added_paths:
+            continue
+        add_header(path = header.path, visibility = "", can_compile = False)
+        added_paths.add(header.path)
+
```

Comment:
note that the core difference is below the paths are already strings, where here we are processing file objects in a context where their paths can be stripped by bazel